boto/connection.py

```diff
@@ -669,9 +669,9 @@ def handle_proxy(self, proxy, proxy_port, proxy_user, proxy_pass):
         if 'http_proxy' in os.environ and not self.proxy:
             pattern = re.compile(
                 '(?:http://)?'
```

Comment:
Why do we not have an `r` marked here ?

Reply:
As there is no backslash character (\) in the string. warnings are raised only for an invalid backslash character.

Reply:
Same comment as for the other such case.